Support closest-corner/farthest-corner in circle and ellipse shapes

smfr opened the pull request:
#### d3dd2bfb693614ceecd21241a5036baab651877a
<pre>
Support closest-corner/farthest-corner in circle and ellipse shapes
<a href="https://bugs.webkit.org/show_bug.cgi?id=277454">https://bugs.webkit.org/show_bug.cgi?id=277454</a>
<a href="https://rdar.apple.com/132936677">rdar://132936677</a>

Reviewed by Tim Nguyen.

Add support for the `closest-corner` and `farthest-corner` radial-size keywords to the `circle()` and `ellipse()` basic
shapes.

This is specified in <a href="https://drafts.csswg.org/css-shapes/">https://drafts.csswg.org/css-shapes/</a> and <a href="https://drafts.csswg.org/css-images-4/#radial-size.">https://drafts.csswg.org/css-images-4/#radial-size.</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-circle-closest-corner-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-circle-closest-corner.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-circle-farthest-corner-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-circle-farthest-corner.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-ellipse-closest-farthest-corner-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-ellipse-closest-farthest-corner.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/reference/clip-path-ellipse-2-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-valid-expected.txt:
* Source/WebCore/css/BasicShapeConversion.cpp:
(WebCore::basicShapeRadiusToCSSValue):
(WebCore::cssValueToBasicShapeRadius):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeShapeRadius):
* Source/WebCore/rendering/style/BasicShapes.cpp:
(WebCore::BasicShapeCircle::floatValueForRadiusInBox const):
(WebCore::BasicShapeEllipse::floatSizeForRadiusInBox const):
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/BasicShapes.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/281808@main">https://commits.webkit.org/281808@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/65e11156b60c4a5f97caf2df9ab6326be580e1cf

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61063 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40424 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13644 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65000 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11612 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/63193 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48101 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11887 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/49360 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/8069 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/63097 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/37619 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52903 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/30190 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/34298 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/10127 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10524 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56116 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/10423 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66730 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5010 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/10238 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56730 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5032 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52866 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/56923 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13623 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4151 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/36228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37311 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38405 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/37055 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-ellipse-closest-farthest-corner-expected.html:
```diff
@@ -0,0 +1,21 @@
+    </style>
+</head>
+<body>
+	<p>The test passes if there is a full green ellipse.</p>
```
nit: bad indent

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-ellipse-closest-farthest-corner-expected.html:
```diff
@@ -0,0 +1,21 @@
+<body>
+	<p>The test passes if there is a full green ellipse.</p>
+    <svg>
+      <ellipse cx="325" cy="200" rx="247.487" ry="125" fill="green"/>
```
inconsistent ident here (2 vs. 4)

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-ellipse-closest-farthest-corner.html:
```diff
@@ -0,0 +1,38 @@
+	<link rel="help" href="http://www.w3.org/TR/css-masking-1/#clipping-paths">
+	<link rel="help" href="http://www.w3.org/TR/css-masking-1/#propdef-clip-path">
+	<link rel="match" href="reference/clip-path-ellipse-2-ref.html">
+    <meta name="fuzzy" content="maxDifference=0-1; totalPixels=0-200">
```
weird indent

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-ellipse-closest-farthest-corner.html:
```diff
@@ -0,0 +1,38 @@
+    </style>
+</head>
+<body>
+	<p>The test passes if there is a full green ellipse.</p>
```
weird indent

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/reference/clip-path-ellipse-2-ref.html:
```diff
@@ -0,0 +1,21 @@
+    </style>
+</head>
+<body>
+	<p>The test passes if there is a full green ellipse.</p>
```
weird indent

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/reference/clip-path-ellipse-2-ref.html:
```diff
@@ -0,0 +1,21 @@
+<body>
+	<p>The test passes if there is a full green ellipse.</p>
+    <svg>
+      <ellipse cx="325" cy="200" rx="247.487" ry="125" fill="green"/>
```
weird indent (2 vs 4)

smfr committed:
Support closest-corner/farthest-corner in circle and ellipse shapes
https://bugs.webkit.org/show_bug.cgi?id=277454
rdar://132936677

Reviewed by Tim Nguyen.

Add support for the `closest-corner` and `farthest-corner` radial-size keywords to the `circle()` and `ellipse()` basic
shapes.

This is specified in https://drafts.csswg.org/css-shapes/ and https://drafts.csswg.org/css-images-4/#radial-size.

* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-circle-closest-corner-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-circle-closest-corner.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-circle-farthest-corner-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-circle-farthest-corner.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-ellipse-closest-farthest-corner-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-ellipse-closest-farthest-corner.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/reference/clip-path-ellipse-2-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/circle-function-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/ellipse-function-valid-expected.txt:
* Source/WebCore/css/BasicShapeConversion.cpp:
(WebCore::basicShapeRadiusToCSSValue):
(WebCore::cssValueToBasicShapeRadius):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeShapeRadius):
* Source/WebCore/rendering/style/BasicShapes.cpp:
(WebCore::BasicShapeCircle::floatValueForRadiusInBox const):
(WebCore::BasicShapeEllipse::floatSizeForRadiusInBox const):
(WebCore::operator<<):
* Source/WebCore/rendering/style/BasicShapes.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/281808@main
